examples/src/main/java/org/apache/spark/examples/BasicAuthFilter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples;
+
+import com.sun.jersey.core.util.Base64;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.StringTokenizer;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+
+/**
+ *  Servlet filter example for Spark web UI.
+ *  BasicAuthFilter provides a functionality of basic authentication.
+ *  The credentials of the filter are "spark-user:spark-password".
+ *
+ *  Usage:
+ *
+ *  $ $SPARK_HOME/bin/spark-submit \
+ *      --jars examples/target/spark-examples_2.10-1.6.0-SNAPSHOT.jar \
+ *      --master <Master URL of your Spark cluster> \
+ *      --conf spark.ui.filters=org.apache.spark.examples.BasicAuthFilter
+ *
+ *  The web UI of your application will ask you username and password.
+ *  Required credentials information is
+ *    username: "spark-user"
+ *    password: "spark-password"
+ */
+public class BasicAuthFilter implements Filter {
```

Comment:
We can make this filter simpler to removing overkill validations and tokenizers. 

Although this filter is not against an actual domain (yet only hard-coded credentials), it was difficult to integrate servlet filter with our Spark environment with only [security documentation](https://spark.apache.org/docs/latest/security.html) for us. 
Therefore I think some type of filter example is helpful anyway. So could you give us an idea, if possible, what kind of scenario can be sufficient as an example for developers who are trying to integrate servlet filter with Spark cluster?
Or writing sample code into documentation might be somewhat helpful.